src/calls/components/CallsView/Calendar.jsx

```diff
@@ -0,0 +1,241 @@
+
+        if (this.state.years.length === 0) {
+            let years = [];
+            for (let y = 2000; y <= getYear(new Date()); y++) {
+                years.push(y);
+            }
+            this.setState({ years: years });
+        }
+    }
+
+    handleOnDateClick = event => {
+        let clickedDay = event.target.textContent;
+        let currentDate = this.state.date;
+        this.setState(
+            {
+                date: addDays(currentDate, clickedDay - currentDate.getDate())
+            },
+            () => this.props.handleChangeTableData(this.state.date)
+        );
+    };
+
+    handleChangeDate = event => {
+        let option = event.target.textContent;
+        if (option === '-') {
+            let currentDate = this.state.date;
+            this.setState({ date: subMonths(currentDate, 1) }, () =>
+                this.props.handleChangeTableData(this.state.date)
+            );
+        } else if (option === '+') {
+            let currentDate = this.state.date;
+            this.setState({ date: addMonths(currentDate, 1) }, () =>
+                this.props.handleChangeTableData(this.state.date)
+            );
+        } else {
+            if (!parseInt(option)) {
+                let currentDate = this.state.date;
+                this.setState(
+                    {
+                        date: setMonth(
+                            currentDate,
+                            this.state.months.indexOf(option)
+                        )
+                    },
+                    () => this.props.handleChangeTableData(this.state.date)
+                );
+            } else {
+                let currentDate = this.state.date;
+                this.setState(
+                    {
+                        date: setYear(currentDate, option)
+                    },
+                    () => this.props.handleChangeTableData(this.state.date)
+                );
+            }
+        }
+    };
+
+    render() {
+        let numOfDays = getDaysInMonth(this.state.date);
+        let startDay = getDay(startOfMonth(this.state.date));
+
+        let blanks = [];
+        let days = [];
+
+        for (let i = 1; i < startDay; i++) {
+            blanks.push(<td key={i + 'b'}></td>);
+        }
+
+        let currentDay = this.state.date.getDate();
+        for (let i = 1; i <= numOfDays; i++) {
+            if (i === currentDay) {
+                days.push(
+                    <td key={i + 'd'} className="text-center btn-success">
+                        {i}
+                    </td>
+                );
+            } else {
+                days.push(
+                    <td key={i + 'd'} className="text-center">
+                        {i}
+                    </td>
+                );
+            }
+        }
+
+        let total = [...blanks, ...days];
+
+        let rows = [];
+        let cells = [];
+
+        total.forEach((td, i) => {
+            if (i % 7 !== 0) {
```

Comment:
Ugh. Can this be handled by CSS, or can we at least [chunk](https://lodash.com/docs#chunk) the array, and iterate through the chunks?

Reply:
What's wrong with this implementation?

Reply:
I replaced it with one for loop which pushing sliced rows by 7 (slice(start, end) function).

Reply:
CSS !?

Reply:
>What's wrong with this implementation?

The code does the correct thing, but it's a custom implementation of a common array manipulation. Same reason why you chose to use `forEach()`. You could have also used a `for` loop. and kept track of the index manually.

>CSS, I don't know how can I do it with that !?

A flex container with 7 items per row.